app/main.py

```diff
@@ -1,6 +1,21 @@
-from typing import Callable
+from functools import wraps
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage = {}
+
+    @wraps(func)
+    def wrapper(*args: Any) -> Callable:
```

Comment:
The return type annotation for the `wrapper` function should be `Any` instead of `Callable`, as the `wrapper` function returns the result of `func(*args)`, which can be of any type, not necessarily a callable.